### ISSUE TITLE
Runtime Error R6016 - Not enough space for thread data

### ISSUE THREAD

**Issue by johngrabner** [desktop/desktop#5356]
## Description

Github desktop v1.3.2 running in background.  This dialog appears about once a day.
Github desktop appears to continue to work.

## Version

* GitHub Desktop: 1.3.2

* Operating system: windows 10

## Steps to Reproduce

just run in background

**Comment by shiftkey** [comment]
@johngrabner thanks for the report!

The best lead I have for this error is [this MSDN documentation link](https://msdn.microsoft.com/en-us/library/d509d9f9.aspx?f=255&MSPPError=-2147217396) but I'm not really sure how it might be happening.

Could you attach the full log files? The **Help** | **Show Logs in Explorer** menu items should open the logs folder, and inside those are the log files generated by day. This will provide some more information to help troubleshoot.

**Comment by johngrabner** [comment]
[2018-08-08.desktop.production.log](https://github.com/desktop/desktop/files/2270722/2018-08-08.desktop.production.log)
[2018-08-07.desktop.production.log](https://github.com/desktop/desktop/files/2270723/2018-08-07.desktop.production.log)
[2018-08-06.desktop.production.log](https://github.com/desktop/desktop/files/2270724/2018-08-06.desktop.production.log)
[2018-08-05.desktop.production.log](https://github.com/desktop/desktop/files/2270725/2018-08-05.desktop.production.log)
[2018-08-04.desktop.production.log](https://github.com/desktop/desktop/files/2270727/2018-08-04.desktop.production.log)

attached are a couple of days of logs

**Comment by johngrabner** [comment]
![github desktop](https://user-images.githubusercontent.com/8209285/43847616-9ed82c04-9af6-11e8-83e5-d15c8dfcbcb8.PNG)
[2018-08-08.desktop.production.log](https://github.com/desktop/desktop/files/2270860/2018-08-08.desktop.production.log)

latest popup and associated logs

**Comment by johngrabner** [comment]
ctrl-c does not show much.  I also did a ctl alt delete and listed the task.

---------------------------
Microsoft Visual C++ Runtime Library
---------------------------
Runtime Error!

Program: C:\Users\john_\AppData\Local\GitHubDesktop\a...

R6016

- not enough space for thread data


---------------------------
OK   
---------------------------


![git2](https://user-images.githubusercontent.com/8209285/43959043-0de91036-9c73-11e8-82e8-1e6c4ea0ff86.PNG)
![git 3](https://user-images.githubusercontent.com/8209285/43959139-6dc354f8-9c73-11e8-8a7b-b93d84f1ebd4.PNG)
![git 4](https://user-images.githubusercontent.com/8209285/43959173-8e90189c-9c73-11e8-8ffc-e46a4c69b911.PNG)

**Comment by johngrabner** [comment]
It looks like "git-remote-https.exe is launch when you click "submit" from the desktop app.
I did a submit with this locked up. I noticed another was launched when I did a submit and it closed when the submit was done. So this may explain why nothing appears to be non-functional.

**Comment by shiftkey** [comment]
@johngrabner are you able to reproduce this with 1.3.3 (which we just published)?

**Comment by johngrabner** [comment]
[cid:3c79e528-2957-4d47-9a5c-a88fea040462]

just got the same message again.

________________________________

**Comment by shiftkey** [comment]
@johngrabner I wonder if it's related to whatever version of the Microsoft Visual C++ Runtime you have installed. I couldn't dig up anything specific to Git for Windows triggering this specific error code in normal use, so maybe running an updated version of this will fix the issue?

**Comment by johngrabner** [comment]
don't do any C++, so no idea how to read what version of C++ Runtime on the machine. If you tell me, i'll look it up.

Is there a way to turn on verbose logging so when this occurs, you may see what is going on?

Can git-remote-https be launched from command line? if so, maybe we can try to isolate it to see it it complains when running on own and not being kicked into action from github descktop?

Anyhow, happy to help, just tell me what to do.

**Comment by shiftkey** [comment]
> don't do any C++, so no idea how to read what version of C++ Runtime on the machine. If you tell me, i'll look it up.

The name is kind of confusing here - it's a runtime that other software relies on, so it's likely an indirect dependency. 

On my machine I have these two versions installed (both version `14.12.25810.0`, but one for x86 and one for x64):

<img width="486" src="https://user-images.githubusercontent.com/359239/44234907-31e76680-a17e-11e8-9c7d-c1d834819c2d.png">

> Can git-remote-https be launched from command line? if so, maybe we can try to isolate it to see it it complains when running on own and not being kicked into action from github descktop?

This is part of [Git for Windows](https://gitforwindows.org) and is a subprogram that `git.exe` invokes when it needs to communicate with a remote server. The actual details of this problem is that `git-remote-https.exe` depends on APIs that are part of `msvcrt.dll`, and this is the version I have at `C:\Windows\System32\msvcrt.dll` that it should be using:

<img width="434"  src="https://user-images.githubusercontent.com/359239/44235127-dcf82000-a17e-11e8-8064-52fdd1ad3139.png">

I'm also on Windows build `10.0.17134.228` with no pending updates - not sure what specific build of Windows 10 you're running (running `cmd /c ver` will print this).

**Comment by shiftkey** [comment]
@johngrabner none of the attached photos came through. Are you able to add the details on the GitHub website?

**Comment by johngrabner** [comment]
I did not restart the PC after upgrading github desktop.
Also had pending windows 10 changes.
After the restart, and overnight, I have not seen this message.
I would not put much on this since the issue was seen for a while when there were no pending changes on windows.  But can't be sure given I have not found a clear trigger.

I have 11 projects with uncommitted changes.  Since the restart, I have only worked on one.  Usually, I have tones of open windows.  It will take a few days to reproduce If the cause was too many open things or running too long. 

Anyhow, here are the pictures that did not make it yesterday.

![image](https://user-images.githubusercontent.com/8209285/44269606-a9f37200-a1fa-11e8-8c1f-52257807aa8e.png)
![image](https://user-images.githubusercontent.com/8209285/44269612-b1b31680-a1fa-11e8-9778-9e77179447ba.png)

I will give it a few days, and if I don't see, i'll close this ticket.
